hnn_core/params.py

```diff
@@ -47,6 +48,46 @@ def __init__(self, params_fname):
                 params[key] = params_input.pop(key)
             self[key] = params[key]
 
+    def read_param_file(self, params_fname):
+        """Read from a .param file (legacy)"""
+        params_input = {}
+        with open(params_fname, 'r') as fp:
+            ln = fp.readlines()
+            for l in ln:
+                s = l.strip()
+                if s.startswith('#'):
+                    continue
+                sp = s.split(':')
+                sp[1] = sp[1].strip()
+                if len(sp[1]) > 0:
+                    if '.' in sp[1] or 'e' in sp[1]:
+                        try:
+                            value = float(sp[1])
+                        except:
+                            value = str(sp[1])
+                    else:
+                        try:
+                            value = int(sp[1])
+                        except:
+                            value = str(sp[1])
+                    params_input[sp[0].strip()] = value
+
+        nprox, ndist = _count_evoked_inputs(params_input)
+
+        # create a copy of params_default through which to iterate
+        params = get_params_default(nprox, ndist)
+
+        for key in params.keys():
+            if key in params_input:
+                params[key] = params_input.pop(key)
+            self[key] = params[key]
```

Comment:
what if I read the entire file and add the curly braces and comas to turn it into a standard json on the fly? That way you ensure that both ways of reading the file will give you consistent results

Reply:
not in favor b/c reading the proper way is so easy.